core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -75,6 +75,8 @@ class TaskInfo(
   }
 
   private[spark] def markFinished(state: TaskState, time: Long = System.currentTimeMillis) {
+    // finishTime should be set larger than 0, otherwise "finished" below will return false.
+    assert(time != 0)
```

Comment:
Nit: shouldn't you then assert that `time > 0` given the comment? do you really mean to assert this, or is it an argument check?